### DISCUSSION TITLE
Junk at end of input

### DISCUSSION THREAD

**Discussion by Rouneq** [kannibalox/pyrosimple#65] [Q&A] [answered]
This might not quite belong here, but I'm working off the examples set in full-example.rc, so I'm hoping I'm not too far afield in my question.

Previously, I was setting up a watch schedule for torrents dumped into a specific directory. The paths were all hard-coded. Like so:

```
# Watch Scheduling
schedule2 = watch_freeleech,1,2,"load.start=/media/0cfa/<username>/private/rtorrent/watch/freeleech/*.torrent,d.directory.set=/media/0cfa/<username>/private/rtorrent/freeleech,d.delete_tied=,d.custom1.set=Freeleech"
```

I thought, "Hey! I can use that nifty `cat` action to do some clean-up/shortening!" And came up with the modified line:

```
# Watch Scheduling
schedule2 = watch_freeleech,1,2,"load.start=(cat,(cfg.watch),\"freeleech/*.torrent\"),d.directory.set=(cat,(cfg.basedir),\"freeleech/\"),d.delete_tied=,d.custom1.set=Freeleech"
```

Starting rtorrent, the UI opens, but then I see repeats of

`Scheduled command failed: watch_freeleech: Junk at end of input.`

I've double-checked the config file and there is nothing after `schedule2` line. Any thought on what I messed up here? Or a better way to accomplish what I'm after? For context, the torrent files are placed in this directory by a third-party app telling the server to go fetch the torrent file into there.

**Comment by kannibalox** [comment, marked as answer]
Unfortunately you've fallen down rtorrent's nesting hell. It might technically be possible to do it in one line, but the easiest thing to do is add another `cfg.` method that you can reference without having to use `cat`:

```ini
method.insert = cfg.freeleech_download,    private|const|string, (cat,(cfg.basedir),"freelech")
schedule2 = watch_freeleech,1,2,"load.normal=(cat,(cfg.watch),\"freeleech/*.torrent\"),d.directory.set=$cfg.freeleech_download=,d.delete_tied=,d.custom1.set=Freeleech"
```
